docs/source/graphs.rst

```diff
@@ -97,3 +97,39 @@ special and others can write different schedulers better suited to other
 applications or architectures easily.  Systems that emit dask graphs (like
 Dask Array, Dask Bag, and so on) may leverage the appropriate scheduler for
 the application and hardware.
+
+
+Task Expectations
+-----------------
+
+When a task is submitted to Dask for execution, there are a number of assumptions
+that are made about that task. In general, tasks with side-effects that alter the
+state of a future in-place are not recommended. Dask Futures are mutable, and can
```

Comment:
So the term future here is a little bit off.  I might suggest saying "Values stored by Dask are mutable".  Semantically a Future is similar to a pointer.  It refers to a remotely held value.  Futures themselves are actually immutable-ish (changing the pointer value is undefined), it's that the data that they point to is not protected in any way which is the problem.

Reply:
Yes, good point. I was sloppy with my wording there.